### PR TITLE
github action for integration pipeline

### PR DESCRIPTION
**Pull request by kevinrue**

### CODE REVIEW

**Comment by kevinrue** [comment]
> The jax pins are also rather sus. If you're worried about the chex failure, you need to wait for the next scvi-tools release

I'll be more than happy to implement a cleaner fix when possible, but jax is the first thing that broke.
https://github.com/DendrouLab/panpipes/actions/runs/7831999626/job/21369644762#step:7:52

Then, restoring the last working version of jax threw an error with numpy, hence the second pin.

Happy to hear about any other fix that makes the action work right now. However, I did what had to be done to be able to work on the action itself so that it runs the pipeline.

**Comment by Zethson** [comment]
Yeah if you look at the CI log again you'll see that scvi-tools is the fault. You can install it from main for now or simply wait for 1-2 weeks where it'll magically fix itself

**Comment by kevinrue** [comment]
Ah ok. I must have missed the scvi-tools bit then. Maybe we open an issue as a sticky note to clean this up in 1-2 weeks?

At the moment, I just need a working environment to test the pipeline and demonstrate that it runs in a github action. I agree that on the long run, we want to set up the testing "the right way" with the latest version of all dependencies.

**Comment by Zethson** [comment]
@kevinrue whatever works for you :) I just wanted to provide a quick perspective of mine

File: .github/workflows/integration-ci.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  debug: 'true'
+
+jobs:
+  integration:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest"] # , "macos-latest", "windows-latest"
+        python-version: ["3.9"]
```
[the remark this answers is not in the document]
Comment (bio-la) [reply .github/workflows/integration-ci.yml:21]: agreed, can you please test the CI with 3.11 @kevinrue ? thanks!

**Review by bio-la — CHANGES_REQUESTED** [review]
test python=3.11 @kevinrue and if all works we can safely merge. 
we give ourselves a couple of weeks to figure dependencies

File: pyproject.toml [hunk 2]
```diff
@@ -35,14 +35,16 @@ dependencies = [
...
+    "jax<=0.4.23",
+    "jaxlib<=0.4.23",
     "leidenalg", 
     "louvain",
     "matplotlib<=3.7.3",
     "mofapy2",
     "mudata>=0.2.1",
     "muon",
-    "numpy>=1.22.4", 
+    "numpy==1.22.4",
```
[the remark this answers is not in the document]
Comment (kevinrue) [reply pyproject.toml:47]: see https://github.com/DendrouLab/panpipes/issues/186

File: .github/workflows/integration-ci.yml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+env:
+  debug: 'true'
```
[the remark this answers is not in the document]
Comment (kevinrue) [reply .github/workflows/integration-ci.yml:12]: Agreed. Even the 'debug' steps are quite useful. I just wanted a way to highlight and toggle them off if desired. Can revisit later. Low priority / no impact for now.

**Comment by kevinrue** [comment]
@bio-la we're back on track and i think ready to merge this one finally now

File: panpipes/python_scripts/batch_correct_scvi.py [hunk 4]
```diff
@@ -159,7 +159,7 @@
 vae.history["elbo_train"].plot()
 plt.savefig(os.path.join(args.figdir, "scvi_elbo_train.png"))
 
-fig, axs = plt.subplots(nrows=3, ncols=4, figsize=(16,8))
+fig, axs = plt.subplots(nrows=4, ncols=4, figsize=(16,10))
```
Comment (bio-la) [inline panpipes/python_scripts/batch_correct_scvi.py:162]: this will need formatting with `f.delaxes(ax[4,XX])`

**Review by bio-la — APPROVED** [review]

File: pyproject.toml [hunk 5]
```diff
@@ -26,43 +26,43 @@ classifiers = [
     "Programming Language :: R"
     ]
 
-requires-python = ">= 3.9"
+requires-python = ">= 3.11"
 
 dependencies = [
```
Comment (Zethson) [inline pyproject.toml:31]: Just a minor comment, but you don't need to specify transitive dependencies -> anndata, jax etc are already included in e.g. scvi-tools. Only if you need to pin them.

Moreover, dependencies such as `pytest` are not runtime dependencies and should be moved into a dev section

Comment (kevinrue) [reply pyproject.toml:31]: you're making a lot of good points, but can we open those as issues, as they will otherwise get lost as comments

**Pull request merged**